wikipedia/track.py

```diff
@@ -0,0 +1,108 @@
+import csv
+import math
+import random
+import re
+from os import getcwd
+from os.path import dirname
+from typing import Iterator
+
+from esrally.track.params import ParamSource
+
+QUERIES_DIRNAME: str = dirname(__file__)
+QUERIES_FILENAME: str = f"{QUERIES_DIRNAME}/queries.csv"
+
+SEARCH_APPLICATION_ROOT_ENDPOINT: str = "/_application/search_application"
+
+
+def query_iterator(k: int) -> Iterator[str]:
+    with open(QUERIES_FILENAME) as queries_file:
+        csv_reader = csv.reader(queries_file)
+        next(csv_reader)
+        queries_with_probabilities = list(tuple(line) for line in csv_reader)
+
+        queries = [query for query, _ in queries_with_probabilities]
+        probabilities = [float(probability) for _, probability in queries_with_probabilities]
+
+        for query in random.choices(queries, weights=probabilities, k=k):
+            # remove special chars from the query + lowercase
+            yield re.sub("[^0-9a-zA-Z]+", " ", query).lower()
```

Comment:
It might make sense to compile the regexp once outside of the function even like `regexp=re.compile("[^0-9a-zA-Z]+")` and just `yield regexp.sub(" ", query).lower()` here from a performance view. (Some evidence of performance benefits [here](https://medium.com/geoblinktech/so-a-few-months-ago-i-had-to-search-the-quickest-way-to-apply-a-regular-expression-to-a-huge-c0883f8d4e4f)).

Reply:
I did a change a while ago to compile the regexp only once in a global constant.